localstack-core/localstack/services/lambda_/invocation/execution_environment.py

```diff
@@ -274,6 +275,14 @@ def keepalive_passed(self) -> None:
             self.id,
             self.function_version.qualified_arn,
         )
+        with self.status_lock:
```

Comment:
docs: Would it make sense to add some explanation about the motivation why this is needed to guard against the race condition?

Reply:
Agreed, added!